### ISSUE TITLE
[Feature Request] Folder Compare between Commits in external diff tools

### ISSUE THREAD

**Issue by LehMichael** [ForkIssues/TrackerWin#28]
It would be nice to be able to compare between two commits with folder compare
(i.e. with Beyond Compare).

**Comment by jerone** [comment]
Agree. With Devart Code Compare GUI.

**Comment by DanPristupov** [comment]
How is that possible technically? Should Fork copy both old and new repository states to a temp directory and then pass the arguments to the diff tool?

**Comment by jerone** [comment]
I guess yes. 

Here is the docs for Devart Code Compare: https://www.devart.com/codecompare/docs/index.html?comparing_via_command_line.htm

**Comment by LehMichael** [comment]
This is even a feature of git. See --dir-diff Option here:
https://git-scm.com/docs/git-difftool

**Comment by LehMichael** [comment]
Now that custom actions are a thing, this is pretty easy to implement yourself!

#### Custom Action 1:
**Title:**  Select for Compare
**Script:**
1.bat:
`echo %2 > %TEMP%\%1.txt`
**Parameters:** `$reponame $sha`

#### Custom Action 2:
**Title:**  Compare with selected
**Script:**
2.bat:
`set /p commit=< %TEMP%\%1.txt`
`%3 difftool -d %commit% %2`
**Parameters:** `$reponame $sha $git`

For this to work you need to configure your difftool first. For beyond compare execute the following lines in the git shell:
`git config --global diff.tool bc3`
`git config --global difftool.bc3.path "c:/program files/beyond compare 4/bcomp.exe"`
`git config --global difftool.bc3.trustExitCode true`

For other tools i don't know, but a quick google search should help you out!

**Comment by AirQuick** [comment]
> How is that possible technically? Should Fork copy both old and new repository states to a temp directory and then pass the arguments to the diff tool?

I guess you don't have to do that.
In SourceTree, Custom Action parameter `$SHA` represents "the hash of the selected commit(s)". Note "(s)". If you have a custom action to run `git` with `difftool --dir-diff $SHA`, you can compare two commits by selecting the commits and invoking the custom action.
Could you consider doing a similar variable expansion in Fork?

**Comment by michael-hawker** [comment]
Yeah, would be nice to see a nice menu option exposed for 'Compare' when having right-clicking on the non-active branch or with a remote.

It'd just have to call the built-in git difftool command with whatever git.config settings are setup (#547).

**Comment by borislip** [comment]
As an alternative, it would be nice to see a new custom command target, which operates on a selection of 2 commits, and allows defining your own commands with both SHAs available, this way we could add "git difftool -d $SHA1 $SHA2" as a custom command. In the meanwhile, i actually use the following as a workaround:

Bat file:
```@echo off
if not "%1"=="0" echo %1>git_diff.left
if not "%2"=="0" echo %2>git_diff.right
if exist git_diff.left set /p left=<git_diff.left
if exist git_diff.right set /p right=<git_diff.right
if not "%left%"=="" ( if not "%right%"=="" (
git difftool -d %left% %right%
del git_diff.left;git_diff.right
))
exit 0
```

Custom command 1 (select left side):
`git_diff.bat $SHA 0`

Custom command 2 (select right side):
`git_diff.bat 0 $SHA`

Edit: just noticed this issue is from 2018... guess it means this feature isn't gonna happen :(
